torch/distributions/weibull.py

```diff
@@ -0,0 +1,50 @@
+from numbers import Number
+import math
+import torch
+from torch.distributions import constraints
+from torch.distributions.uniform import Uniform
+from torch.distributions.transformed_distribution import TransformedDistribution
+from torch.distributions.transforms import AffineTransform, ExpTransform, PowerTransform
+from torch.distributions.utils import _finfo, broadcast_all
+from torch.distributions.gumbel import euler_constant
+
+
+class Weibull(TransformedDistribution):
+    r"""
+    Samples from a two-parameter Weibull distribution.
+
+    Example:
+
+        >>> m = Weibull(torch.tensor([1.0]), torch.tensor([1.0]))
+        >>> m.sample()  # sample from a Weibull distribution with scale=1, concentration=1
+        tensor([ 0.4784])
+
+    Args:
+        scale (float or Tensor): Scale parameter of distribution (lambda).
+        concentration (float or Tensor): Concentration parameter of distribution (k/shape).
+    """
+    arg_constraints = {'scale': constraints.positive, 'concentration': constraints.positive}
+    support = constraints.positive
+
+    def __init__(self, scale, concentration, validate_args=None):
+        self.scale, self.concentration = broadcast_all(scale, concentration)
+        self.concentration_reciprocal = self.concentration.reciprocal()
+        finfo = _finfo(self.scale)
+        base_dist = Uniform(self.scale.new(self.scale.size()).fill_(finfo.tiny), 1 - finfo.eps)
```

Comment:
Is there a reason you're using `log(Uniform)` rather than `Exponential`? It might be cheaper to use `Exponential` which calls the PyTorch exponential generator under the hood. This would also reduce the number of transforms from 4 to 2.
```py
base_dist = Exponential(self.scale.new_tensor(self.scale.size()).fill_(1.0))
transforms = [PowerTransform(exponent=self.concentration_reciprocal),
              AffineTransform(loc=0, scale=self.scale)]
```